docs/user_guide.md

```diff
@@ -665,7 +665,9 @@ BENCHMARK_TEMPLATE_DEFINE_F(MyFixture, DoubleTest, double)(benchmark::State& st)
   }
 }
 
+/* DoubleTest is NOT registered */
```

Comment:
I *think* `//`-style comments are used elsewhere, so let's be consistent.
I think it would be good to also mention `IntTest`,
is `BENCHMARK_TEMPLATE_F` enough, and only the `BENCHMARK_TEMPLATE_DEFINE_F`
needs to be followed up with `BENCHMARK_REGISTER_F` or not?

Reply:
👍, I'm happy to do that (and will write some tests to confirm!).

In the user guide, [Fixtures](https://github.com/google/benchmark/blob/main/docs/user_guide.md#fixtures) is similar and uses `/* ... */` comments - I think it's within scope to update that one.

Shall I update [the one other instance in Preventing Optimization](https://github.com/google/benchmark/blob/main/docs/user_guide.md#preventing-optimization) as well?

Reply:
> 👍, I'm happy to do that (and will write some tests to confirm!).
 
> In the user guide, [Fixtures](https://github.com/google/benchmark/blob/main/docs/user_guide.md#fixtures) is similar and uses `/* ... */` comments - I think it's within scope to update that one.
> 
> Shall I update [the one other instance in Preventing Optimization](https://github.com/google/benchmark/blob/main/docs/user_guide.md#preventing-optimization) as well?

Eh, i guess why not?

Reply:
Done. PTAL.